worlds/sc2wol/PoolFilter.py

```diff
@@ -18,73 +20,102 @@
 FACTORY_UNITS = {"Hellion", "Vulture", "Goliath", "Diamondback", "Siege Tank", "Thor", "Predator", "Widow Mine"}
 STARPORT_UNITS = {"Medivac", "Wraith", "Viking", "Banshee", "Battlecruiser", "Hercules", "Science Vessel", "Raven", "Liberator", "Valkyrie"}
 
-PROTOSS_REGIONS = {"A Sinister Turn", "Echoes of the Future", "In Utter Darkness"}
 
-
-def filter_missions(multiworld: MultiWorld, player: int) -> Dict[int, List[str]]:
+def filter_missions(multiworld: MultiWorld, player: int) -> Dict[MissionPools, List[SC2Mission]]:
     """
     Returns a semi-randomly pruned tuple of no-build, easy, medium, and hard mission sets
     """
 
     mission_order_type = get_option_value(multiworld, player, "mission_order")
     shuffle_no_build = get_option_value(multiworld, player, "shuffle_no_build")
-    shuffle_protoss = get_option_value(multiworld, player, "shuffle_protoss")
+    enabled_campaigns = get_enabled_campaigns(multiworld, player)
+    disabled_campaigns = get_disabled_campaigns(multiworld, player)
     excluded_missions = get_option_value(multiworld, player, "excluded_missions")
```

Comment:
I noticed excluded missions are responsible for a lot of lookup calls & string comparisons. Of course the YAML option can't be changed, but there are a lot of places where names are taken from SC2Mission values for comparison with excluded missions and also some straight lookups of entire mission lists/sets. Since both are done regularly, it'd be better to do the lookup once at the start and then work with SC2Mission values in the rest of the PoolFilter code, eg.:
```
excluded_mission_names = get_option_value(multiworld, player, "excluded_missions")
excluded_missions = [lookup_name_to_mission[name] for name in excluded_mission_names]
```

Reply:
Done